xbmc/video/VideoDatabase.cpp

```diff
@@ -3153,57 +3153,57 @@ int CVideoDatabase::SetDetailsForSeason(const CVideoInfoTag& details,
 int CVideoDatabase::SetFileForMedia(const std::string& fileAndPath,
                                     VideoDbContentType type,
                                     int mediaId,
-                                    int oldIdFile)
+                                    int oldIdFile,
+                                    const CDateTime& dateAdded,
+                                    int playcount,
+                                    const CDateTime& lastPlayed)
 {
   if ((mediaId < 0 && type != VideoDbContentType::UNKNOWN) || oldIdFile < 0)
     return -1;
 
+  const int idFile{AddFile(fileAndPath)};
+  if (idFile < 0)
+    return -1;
+
+  assert(m_pDB->in_transaction());
+
+  try
+  {
+    // Update here as new file id may already exist (from updating playcount etc.) and AddFile() will not update it
+    m_pDS->exec(PrepareSQL(
+        "UPDATE files SET playCount=%i, lastPlayed='%s', dateAdded='%s' WHERE idFile=%i", playcount,
+        lastPlayed.GetAsDBDateTime().c_str(), dateAdded.GetAsDBDateTime().c_str(), idFile));
+  }
+  catch (...)
+  {
+    CLog::LogF(LOGERROR, " idFile {}, fileAndPath {}, mediaId {}, oldIdFile {} - failed", idFile,
```

Comment:
Why not use the AddFile(url, parentPath, dateAdded, playcount, lastPlayed) overload?
Looks to me like less duplication and more consistent handling of 0 playCount and invalid lastPlayed with the rest of the code.